include/deal.II/lac/solver_gmres.h

```diff
@@ -2079,6 +2199,47 @@ void SolverFGMRES<VectorType>::solve(const MatrixType         &A,
                 SolverControl::NoConvergence(accumulated_iterations, res));
 }
 
+
+
+//----------------------------------------------------------------------//
+
+
+
+template <typename VectorType>
+DEAL_II_CXX20_REQUIRES(concepts::is_vector_space_vector<VectorType>)
+SolverFGMRES<VectorType>::SolverFGMRES(SolverControl            &cn,
+                                       VectorMemory<VectorType> &mem,
+                                       const AdditionalData     &data)
+  : SolverMPGMRES<VectorType>(cn, mem, data)
+{}
+
+
+
+template <typename VectorType>
+DEAL_II_CXX20_REQUIRES(concepts::is_vector_space_vector<VectorType>)
+SolverFGMRES<VectorType>::SolverFGMRES(SolverControl        &cn,
+                                       const AdditionalData &data)
+  : SolverMPGMRES<VectorType>(cn, data)
+{}
+
+
+
+template <typename VectorType>
+DEAL_II_CXX20_REQUIRES(concepts::is_vector_space_vector<VectorType>)
+template <typename MatrixType, typename PreconditionerType>
+DEAL_II_CXX20_REQUIRES(
+  (concepts::is_linear_operator_on<MatrixType, VectorType> &&
+   concepts::is_linear_operator_on<PreconditionerType, VectorType>))
+void SolverFGMRES<VectorType>::solve(const MatrixType         &A,
+                                     VectorType               &x,
+                                     const VectorType         &b,
+                                     const PreconditionerType &preconditioner)
+{
+  LogStream::Prefix prefix("FGMRES");
```

Comment:
Do we still need this `LogStream::Prefix` object? If no, then we could get rid of `solver_internal` and implement everything in `MPGMRES::solve()` directly.

Reply:
I've long thought that we should just do away with `LogStream` everywhere but in the test suite. But if you remove it here, you should probably remove its use in all of the solvers. (And not just remove the prefix, but actually remove writing into `deallog`.)

Reply:
I agree. But let's tackle this in a future pull request.